src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -511,31 +511,64 @@ public static function toHtml5($strString)
 	 * @return string The converted string
 	 *
 	 * @throws \Exception If $strString cannot be parsed
+	 * @throws \InvalidArgumentException If $strString or any token in $arrData contains php code
 	 */
 	public static function parseSimpleTokens($strString, $arrData)
 	{
 		$strReturn = '';
 
-		// Remove any unwanted tags (especially PHP tags)
-		$strString = strip_tags($strString, \Config::get('allowedTags'));
+		// Do not allow php code for security reasons.
+		$containsPhp = function($value) {
+			return strpos($value, '<?php') !== false
+				|| strpos($value, '<?') !== false
+				|| strpos($value, '<%') !== false
+				|| strpos($value, '<%=') !== false
+				|| strpos($value, '<script language="php">') !== false
+				|| strpos($value, '<script language=\'php\'>') !== false;
+		};
+
+		// Validate subject string
+		if ($containsPhp($strString))
+		{
+			throw new \InvalidArgumentException('PHP Code is not allowed in subject string.');
+		}
+
+		// Validate replacement data
+		foreach ($arrData as $k => $v)
+		{
+			if ($containsPhp($v))
+			{
+				throw new \InvalidArgumentException(sprintf('PHP Code is not allowed in replacement data (Key "%s").', $k));
```

Comment:
But how about using `token_get_all()` to check for PHP tags? If it returns only one token of type `T_INLINE_HTML` it should be safe, because it didn’t find an opening tag.